### DISCUSSION TITLE
Revalidate getStaticPaths

### DISCUSSION THREAD

**Discussion by carlesba** [vercel/next.js#40392] [Help]
I'm trying to apply the [Incremental Static Regeneration](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration#) but I couldn't find a way to create or delete resources for a path. This means that I can rebuild a page with some info changes but I can't update the list of available pages when I create/delete resources in my system.

I guess this could work if we could revalidate the getStaticPaths function, but I didn't find a solution. Is there any?

**Comment by icyJoseph** [comment]
Perhaps you can use the [on-demand revalidate function](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation), which would tell next.js to revalidate a set of pages. If you've removed these from your CMS, you could then return `{notFound: true}` when they are regenerated.

1. Build app, paths from getStaticPaths go live
2. Delete on your CMS
3. Call the on-demand revalidation as soon as 2 completes
4. getStaticProps runs on those pages generated on step 1, and those generated afterwards if fallback was `true` or `'fallback'`
5. getStaticProps would fail to find data for these pages because they don't exist on the CMS, so instead of `{props:{...}}`, returns `{notFound: true}`

Sounds like that should work right? https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation

**Reply by Rickyc81** [reply]
I've the same problem, when I delete a post on my CMS, revalidate is triggered, but the static file remain on the server folder..

**Reply by Rickyc81** [reply]
> 

Excuse me, with this procedure, after I delete post from cms, it will be deleted from static html folder?

**Reply by icyJoseph** [reply]
I am not sure, I wanna say no, because the ISR process don't modify the output, as far as I remember.

I guess memory might be a problem, if that's the case, because you'd be storing an old, stale page, and a new page, potentially causing memory hogging.

**Reply by BenLebich** [reply]
This looks like a pretty solid answer. Regarding fallback, I didn't know that existed and step #4 wasn't clear what is fallback for.
This step is critical for creating new pages after the build. Here's the documentation link for anyone who might have gotten stuck at that step:
https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking

`If fallback is 'blocking', new paths not returned by getStaticPaths will wait for the HTML to be generated, identical to SSR (hence why blocking), and then be cached for future requests so it only happens once per path.`

**Comment by carlesba** [comment]
A possible solution might be to use a vercel's hook to trigger a redeployment when a resource is created/deleted. This means we use 2 endpoints:
- The ISR endpoint is called when a resource has changed
- The hook endpoint is called when a resource has been created/deleted.

That should make the trick

**Reply by BenLebich** [reply]
That defeats the whole purpose of ISR. The goal is to minimize redeploys when content is modified/created/deleted.
If I have 5k+ pages (or millions as Vercel claims it scales too) it becomes more expensive to rebuild.

**Reply by carlesba** [reply]
Agreed, but it solves the problem. I didn't find a better solution yet
